src/shared/mkfs-util.c

```diff
@@ -90,6 +90,7 @@ int make_filesystem(
                 const char *node,
                 const char *fstype,
                 const char *label,
+                const char *root,
                 sd_id128_t uuid,
                 bool discard) {
```

Comment:
adding this new argument might be OK, but then this function really should fail on file systems where we don't support the concept yet when set. i.e. return EOPNOTSUPP on those other file systems.

That said, it might be worth adding support for this right away for ext4 and btrfs, given it's simple. mke2fs has a "-d" switch now, mkfs.btrfs has a "-r" switch. None of the others we have explicit support for appears to support a similar swtich.

Reply:
(I understand you don't intend to use that for other file systems. so either add it anyway for completeness sake, or just return EOPNOTSUPP if the root dir is specified but we can't pass it along).

Thing is: label/uuid/discard are all just decoration, but the root dir is more than that, it's *contents* hence I am pretty sure we should not just ignore it if given and we can#t apply it.